spec/models/user_spec.rb

```diff
@@ -35,6 +35,12 @@
       user2.save
       expect(user2).to be_valid
     end
+
+    it "disables validations when reset password" do
+      user = Fabricate.build(:user, email: '')
+      user.save(validate: false)
+      expect(user.reset_password('Password123', 'Password123')).to be_truthy
+    end
   end
 
```

Comment:
Would be nice to have a negative test case, to make sure password validations still prevent you from saving a password that is too short for example when using User#reset_password.

also reset_password is fine, but I might go with set_password because reset brings to my mind a slightly different idea of what this is doing.